components/raftstore/src/store/worker/read.rs

```diff
@@ -441,25 +449,35 @@ where
     C: ProposalRouter<E::Snapshot> + CasualRouter<E>,
     E: KvEngine,
 {
-    fn get_engine(&self) -> &E {
-        &self.kv_engine
+    fn get_tablet(&self, region_id: u64) -> E {
+        self.factory.open_tablet_cache_latest(region_id).unwrap()
     }
 
-    fn get_snapshot(&mut self, create_time: Option<ThreadReadId>) -> Arc<E::Snapshot> {
+    fn get_snapshot(
+        &mut self,
+        create_time: Option<ThreadReadId>,
+        mut region_id: u64,
+    ) -> Arc<E::Snapshot> {
+        if self.factory.get_factory_version() == TabletFactoryVersion::Single {
```

Comment:
I don't see the point of this check. If it's single rocksdb, it can just ignore `region_id`.

Reply:
Now, open_tablet_cache_any of KvEngineFactory does not ignore the region_id. Ok, maybe I should change it rather than check here.